graphstorm-processing/scripts/script_utils.py

```diff
@@ -0,0 +1,213 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import argparse
+from ast import literal_eval
+from typing import Any, Dict, Sequence, Optional
+
+from botocore.config import Config
+import boto3
+
+
+def check_if_instances_available(instance_type: str, instance_count: int, region: str) -> None:
+    """
+    Checks if we have enough of a quota for the requested instance type and count in a region.
+    Raises a runtime error if the requested capacity exceeds the current quota.
+    """
+    quota_name = f"{instance_type} for processing job usage"
+    quota_value = get_quota_value("sagemaker", quota_name, region)
+
+    if int(quota_value) < instance_count:
+        raise RuntimeError(
+            f"Requested {instance_count=} goes"
+            f" above current ({quota_value=}) for {instance_type=}"
+        )
+
+    # TODO: Use sagemaker_client.list_processing_jobs and
+    # describe_processing_job to get current utilization
+
+
+def get_quota_value(service_code: str, quota_name: str, region: str) -> float:
+    """
+    For a given service code, quota name, and region, return the quota value.
+    """
+
+    def get_value_for_quota_name(
+        quotas: Sequence[Dict[str, Any]], quota_name: str
+    ) -> Optional[float]:
+        for quota in quotas:
+            if quota["QuotaName"] == quota_name:
+                return quota["Value"]
+        return None
+
+    quota_client = boto3.client(
+        "service-quotas",
+        config=Config(
+            region_name=region, connect_timeout=5, read_timeout=60, retries={"max_attempts": 20}
+        ),
+    )
+    quota_paginator_client = quota_client.get_paginator("list_service_quotas")
+
+    quota_response = quota_paginator_client.paginate(
+        ServiceCode=service_code,
```

Comment:
By calling the quota service, we will get the Quota data for all possible choices under the specific service? Can we pass-in the quota name?

Reply:
We need the quota code here, so we need to first list all quotas and extract the code from the name. It's how the [boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/service-quotas/client/get_service_quota.html) also recommend this is done.

I'm not sure if there's a more straight-forward solution.